pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1273,12 +1273,13 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                     return null;
                 }
 
+                // Do not print stack traces for expected exceptions
                 if (cause instanceof NoSuchElementException) {
                     cause = new TopicNotFoundException("Topic Not Found.");
```

Comment:
I think we could also try to not create a 2nd exception here which is only used to do `getMessage()`

Reply:
Sure, although we also use the exception here: `BrokerServiceException.getClientErrorCode(cause)`.

Reply:
@merlimat - would you like me to introduce more complicated logic for getting the client error code, or is this sufficient?